src/eko/evolution_operator/grid.py

```diff
@@ -233,15 +233,20 @@ def generate(self, q2):
 
         # integrate matching conditions
         # exact inverse depends on q2 so need to be computed separately
+        # Assume that intrinsic always depends on the scale ( TODO: not correct for the moment but
+        # will be true when we consider the logs )
         is_backward = bool(path[-1].q2_from > path[-1].q2_to)
-        if len(path) > 1 and len(self.ome_members) == 0:
-            if is_backward is False:
+        if len(path) > 1 and self.ome_members is None:
+            if (
+                self.config["intrinsic_range"] is not None
+                or self.config["backward_inversion"] == "exact"
+            ):
+                ome = OperatorMatrixElement(self.config, self.managers)
+            elif is_backward is False:
                 self.config["backward_inversion"] = None
```

Comment:
Changing `self.config` is weird and might be wrong:
- the configs should be the ones set by the user, so you can think about replacing that information by passing it directly as an argument to the function `self.compute_matching_coeffs()`
- in particular this will be an actual bug if your grid contains some forward and some backward Q2s: the first forward Q2 will replace the value for `self.config` forever, and the moment you try to compute a backward one you will not fall in any valid branch but in `ValueError`

Reply:
Okay, I agree.